src/griffe/dataclasses.py

```diff
@@ -176,6 +176,8 @@ def __init__(
         annotation: str | Expr | None = None,
         kind: ParameterKind | None = None,
         default: str | Expr | None = None,
+        docstring: Docstring | None = None,
+        parent: Module | Class | None = None,
```

Comment:
Our `parent` is most definitely a Function here. Maybe we could rename `parent` to `function` to make this more clear and avoid any ambiguity?

Reply:
At the moment, the `parent` is a `Class`, because we are transforming an "Attribute".

Reply:
But the parameter is attached to the `__init__` function that the dataclasses extension generates for the dataclass, right?

First, Griffe parses the InitVar as a class attribute. Then, the dataclasses extension handles the dataclass by re-creating an `__init__` Function and creating it's parameters based on InitVars and else. The docstring of the attribute is passed to the corresponding parameter, then the InitVar attribute is removed. I think it works that way, or at least it should work that way. I'll check the code again.

In the end, the parent of a parameter can only ever be a Function, not a Class 🤔

Reply:
```python
from griffe.tests import temporary_visited_package

code = """
from dataclasses import dataclass, InitVar

@dataclass
class Base:
    a: int
    "Parameter a"
    b: InitVar[int] = 3
    "Parameter b"
"""

with temporary_visited_package("package", {"__init__.py": code}) as module:
    base = module["Base"]
    param_a = base.parameters[1]
    param_b = base.parameters[2]

type(param_a.parent), type(param_b.parent)
# (griffe.dataclasses.Class, griffe.dataclasses.Class)
```


The `__init__` function is [created](https://github.com/has2k1/griffe/blob/371318190ecfc5329f020a52e0f156dd91031031/src/griffe/extensions/dataclasses.py#L176-L187) after the parameters. 

At [member.parent](https://github.com/has2k1/griffe/blob/371318190ecfc5329f020a52e0f156dd91031031/src/griffe/extensions/dataclasses.py#L129), [member](https://github.com/has2k1/griffe/blob/371318190ecfc5329f020a52e0f156dd91031031/src/griffe/extensions/dataclasses.py#L80) is an "Attribute".

Reply:
Same thing in the [Visitor.handle_function](https://github.com/has2k1/griffe/blob/371318190ecfc5329f020a52e0f156dd91031031/src/griffe/agents/visitor.py#L362-L387).

Reply:
A solution would be:
1. create the function without the parameters
2. create the parameters and pass the function as the parent
3. set the parameters attribute on the function

```python
function = Function(..., parent=None)
function.parameters = Parameters([Parameter(..., parent=function) ...])
```

Reply:
Right. Then yes, we could do as you suggest, or do something a bit less pure and assign the function to the parameters when we pass them, just like we do when assigning a member to an object:

```python
# in Function.__init__
for parameter in parameters:
    parameter.function = self
```